New documentation (#394)
AequilibraE/aequilibrae: docs/source/development/softwaredevelopment.rst

    * Unit testing
    * Build/packaging tests
    * Documentation building test
* If the tests pass, then a manual pull request can be approved to merge into develop.

> * The master ande velop branches are protected and therefore can only be written to after the code has been reviewed and approved.

Reviewer: ```suggestion
* The master ande velop branches are protected and therefore can only be written to after the code has been reviewed and approved.
```
```suggestion
* The master and develop branches are protected and therefore can only be written to after the code has been reviewed and approved.
```

AequilibraE testing is done with three tools:

* `Flake8 <https://pypi.org/project/flake8/>`_, a tool to check Python code style
* `Black <https://black.readthedocs.io/en/stable/index.html/>`_, The uncompromising code formatter

Testing is done for Windows, MacOs and Ubuntu Linux on all supported Python versions, and we use GitHub Actions

> to run these tests. These tests need to pass in order to somebody

Reviewer: ```suggestion
to run these tests. These tests need to pass and additionally somebody has to
```
